### PR TITLE
Replaced function insertSyntax

### PR DESCRIPTION
**Pull request by alexriss**
I have had problems with the insertSyntax function. It worked only if no text edit was done.

Now I have rewritten the function. It looks similar to dokuwiki's own n insertTags function.

It even allows you to mark text, which then will be used as the image title.

### CODE REVIEW

**Comment by micgro42** [comment]
With which browsers have you seen this error?

File: src/upload.js [hunk 1]
```diff
@@ -246,14 +246,32 @@ jQuery(function () {
...
+        opts = { nosel: true };
+        text = open + text + close;
+        pasteText(selection,text,opts);
+
+
+
+
+
+
+
```
Comment (micgro42) [inline src/upload.js]: What is the motivation behind all this empty lines?

**Comment by alexriss** [comment]
Chrome on Windows. No error, just nothing happens - the link is not included in the text area. There is an open issue by someone else about exactly this behavior.

File: src/upload.js [hunk 2]
```diff
@@ -246,14 +246,32 @@ jQuery(function () {
...
+        var opts;
+
+        // don't include trailing space in selection
+        if(text.charAt(text.length - 1) == ' '){
+            selection.end--;
+            text = selection.getText();
+        }
+
+        if(text){
+            text = '|' + text;  // use text as image label
```
Comment (micgro42) [inline src/upload.js]: This is not primarily intended for images, but for documents in general. Could you adjust that comment?

**Comment by micgro42** [comment]
For the record: The alternative to merging this would be to replace the calls to `$editarea.text(` with `$editarea.val(`, because `.text(` refers to the original HTML but `.val()` references always the current value. In case of a textarea that would be its content.

However, this pull request provides a nice little extra feature and we should therefore go forward with it.

**Comment by alexriss** [comment]
thanks for the fast response. The caret seems to be working fine in IE11. What does not work, though, is the label for the uploaded file. This is likely because once IE loses focus during drag-and-drop, the selection is unselected. Insertion still works without errors, just without the label-functionality.

**Comment by micgro42** [comment]
Awesome, thanks :)

**Pull request merged**